### DISCUSSION TITLE
How to test if build project is compiling successfully on CI side

### DISCUSSION THREAD

**Discussion by IT-VBFK** [nuke-build/nuke#1101] [Q&A]
Hi

I recently had following situation:
I have implemented some detection for changed files.
To achieve this, I've had to add a parameter and pass `github.event.pull_request.base.ref` into the NUKE build pipeline.
On my local dev device all is running well (with parameter and parameter omitted), in the PR of the project I am contributing to as well.

Now here is the problem: After merging into the `develop` branch `github.event.pull_request.base.ref` was not set to an expected value (had assumed to be `null`), but actually it was an empty string. Thus, the build was failing.

Is there a neat way to previously check all different possibilities and "unit test" the behavior?

**Comment by sandrofigo** [comment]
I'm not yet fully familiar with the API from GitHub so correct me if I'm completely wrong here, but I would assume that `github.event.pull_request.base.ref` is only available when the triggering event is related to a pull request (see https://docs.github.com/en/actions/learn-github-actions/contexts#github-context). Since you stated that you encounter the behavior **after** merging, the merge commit might not be in the context of the original pull request anymore, therefore the base ref is not available.
There might be another property in the `github` context that fits your needs.

**Reply by IT-VBFK** [reply]
Yes.. that is correct.

But my question was, if I can check this **beforehand**, and not have to wait until the PR is merged and then have to see that the pipeline is failing.

Now I know that I have to check for `null` or empty/whitespace string.
